src/ng/rootScope.js

```diff
@@ -892,6 +893,58 @@ function $RootScopeProvider() {
         $browser.$$checkUrlChange();
       },
 
+      /**
+       * @ngdoc method
+       * @name $rootScope.Scope#$suspend
+       * @kind function
+       *
+       * @description
+       * Suspend watchers of this scope subtree so that they will not be invoked during digest.
+       *
+       * This can be used to optimize your application when you know that running those watchers
+       * is redundant.
+       *
+       * **Warning**
+       *
+       * Suspending scopes from the digest cycle can have unwanted and difficult to debug results.
+       * Only use this approach if you are confident that you know what you are doing and have
+       * ample tests to ensure that bindings get updated as you expect.
+       *
+       * Some of the things to consider are:
+       *
+       * * Any external event on a directive/component will not trigger a digest while the hosting
+       *   scope is suspended - even if the event handler calls `$apply` or `$digest`.
+       * * Transcluded content exists on a scope that inherits from outside a directive but exists
+       *   as a child of the directive's containing scope. If the containing scope is suspended the
+       *   transcluded scope will also be suspended, even if the scope from which the transcluded
+       *   scope inherits is not suspended.
+       * * Multiple directives trying to manage the suspended status of a scope can confuse each other:
```

Comment:
Yes good point.  Perhaps a .$isSuspended() getter or similar so a directive can find out whether its scope is already suspended?

Reply:
Hmm this brings up an additional more complex point.

If a parent scope is suspended, but the current scope is not, this scope will not take part in any digests even though it is not officially suspended.

I think that we don't care, since directives that want to suspend a scope don't really care whether an ancestor scope has been suspended. They are only interested in managing their own scope.

The only time this is a problem is if there are multiple directives trying to manage the suspended status of a single scope.

Reply:
Yes, I see the problem. It's a fair assumption that if you asked for $isSuspended() you'd want `true` returned if it was disabled __because of__ a parent, but I think one line in the docs explaining this issue would solve that.

Reply:
Out of interest, how hard would it be to implement a `.$ancestorIsSuspended()`?  

Unfortunately my only case for it is if people use `$scope.$digest` at some point below the `$suspend`'ed element and want to avoid that $digest in those cases. Since you discourage `$scope.$digest`, I am girding myself for rejection here, but it would be useful to me, and `.$ancestorIsSuspended()` would save me writing a custom "recurse up the $parents" function.

Reply:
To find out if an ancestor is suspended, you only need to read the $parent scope until you find one that is suspended.

Reply:
An example is shown [here][1].
@poshest, if you want it to be available on every scope, you can put it on `$rootScope`'s prototype.

[1]: https://github.com/angular/angular.js/pull/16308/files#diff-44d3307b66bf51b52e110307fa57e196R955

Reply:
Awesome! Thanks guys, and for the code example @gkalpak :)